Generals/Code/GameEngine/Source/Common/INI/INI.cpp

```diff
@@ -1903,7 +1906,8 @@ Bool INI::isDeclarationOfType( AsciiString blockType, AsciiString blockName, cha
 		++tempBuff;
 	}
 
-	if (strlen(tempBuff) > blockNameLength) {
+	tempBuffLen = tempBuffEnd - tempBuff;
+	if (tempBuffLen > blockNameLength) {
 		restoreChar = tempBuff[blockNameLength];
 		tempBuff[blockNameLength] = 0;
```

Comment:
Looking at this, this can be simplified too by using `strnicmp`. Writing a null char into the buffer is bogus. Twice in this function.

Then the comment above
```
	// DO NOT RETURN EARLY FROM THIS FUNCTION. (beyond this point)
	// we have to restore the bufferToCheck to its previous state before returning, so
	// it is important to get through all the checks.
```
can also be removed.

And then the early returns can be added too.